File: 2524-largest-positive-integer-that-exists-with-its-negative/largest-positive-integer-that-exists-with-its-negative.java
class Solution {
    public int findMaxK(int[] nums) {
        Arrays.sort(nums);
        int l = 0, r = nums.length - 1;

        while (l < r){
            if (nums[l] > 0 || nums[r] < 0) return -1;
            else if(Math.abs(nums[l]) > nums[r]) l++;
            else if(Math.abs(nums[l]) < nums[r]) r--;
            else return nums[r];
        }return -1;
    }
}

File: 2524-largest-positive-integer-that-exists-with-its-negative/largest-positive-integer-that-exists-with-its-negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        nums.sort()
        print(nums)
        left, right = 0, len(nums) - 1

        while left < right:
            if nums[left] > 0 or nums[right] < 0:
                return -1
            if abs(nums[left]) > abs(nums[right]):
                left += 1
            elif abs(nums[left]) < abs(nums[right]):
                right -= 1
            else:
                return nums[right]
        
        return -1
